src/components/map/MapDirective.js

```diff
@@ -156,6 +156,43 @@ goog.require('ga_styles_service');
                 pitch: Cesium.Math.toDegrees(camera.pitch).toFixed(3)
               });
             }, 1000, false));
+
+            var dereg = [];
+            var setRealPosition = function(itemOrEvt) {
+              var item = (itemOrEvt instanceof ol.Overlay) ? itemOrEvt :
+                  itemOrEvt.element;
+              item.set('realPosition', item.getPosition());
+              item.setPosition();
+              dereg.push(item.on('change:position', function(evt) {
+                var val = evt.target.getPosition();
+                if (val) {
+                  item.set('realPosition', val);
+                  item.setPosition();
+                }
+              }));
+            };
+
+            // Watch when 3d is enabled to show/hide overlays
+            scope.$watch(function() {
```

Comment:
Yes. we have this watch on multiple places maybe we should broadcast an event when 3d is active/deactive.